src/cc/CMakeLists.txt

```diff
@@ -123,41 +116,42 @@ set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${llvm_lib_exclude_f
 set(bcc_common_libs b_frontend clang_frontend
   -Wl,--whole-archive ${clang_libs} ${llvm_libs} -Wl,--no-whole-archive
   ${LIBELF_LIBRARIES})
-set(bcc_common_libs_for_a ${bcc_common_libs} bpf-static)
-set(bcc_common_libs_for_s ${bcc_common_libs} bpf-static)
-set(bcc_common_libs_for_n ${bcc_common_libs})
-set(bcc_common_libs_for_lua b_frontend clang_frontend bpf-static
+set(bcc_common_libs_for_a ${bcc_common_libs})
+set(bcc_common_libs_for_s ${bcc_common_libs})
+set(bcc_common_libs_for_lua b_frontend clang_frontend
   ${clang_libs} ${llvm_libs} ${LIBELF_LIBRARIES})
+if(LIBBPF_FOUND)
+  list(APPEND bcc_common_libs_for_a ${LIBBPF_LIBRARIES})
+  list(APPEND bcc_common_libs_for_s ${LIBBPF_LIBRARIES})
+  list(APPEND bcc_common_libs_for_lua ${LIBBPF_LIBRARIES})
+else()
+  list(APPEND bcc_common_libs_for_a bpf-static)
+  list(APPEND bcc_common_libs_for_s bpf-static)
+  list(APPEND bcc_common_libs_for_lua bpf-static)
+endif()
 
 if(ENABLE_CPP_API)
   add_subdirectory(api)
   list(APPEND bcc_common_libs_for_a api-static)
   # Keep all API functions
   list(APPEND bcc_common_libs_for_s -Wl,--whole-archive api-static -Wl,--no-whole-archive)
-  list(APPEND bcc_common_libs_for_n -Wl,--whole-archive api-static -Wl,--no-whole-archive)
 endif()
 
 if(ENABLE_USDT)
   list(APPEND bcc_api_headers bcc_usdt.h)
   add_subdirectory(usdt)
   list(APPEND bcc_common_libs_for_a usdt-static)
   list(APPEND bcc_common_libs_for_s usdt-static)
-  list(APPEND bcc_common_libs_for_n usdt-static)
   list(APPEND bcc_common_libs_for_lua usdt-static)
 endif()
 
 add_subdirectory(frontends)
 
 # Link against LLVM libraries
-target_link_libraries(bcc-shared ${bcc_common_libs_for_s})
+target_link_libraries(bcc-shared ${bcc_common_libs_for_s} ${LIBBPF_LIBRARIES})
```

Comment:
${LIBBPF_LIBRARIES} is not needed here since ${bcc_common_libs_for_s} has already linked with it, right?

Reply:
Yes indeed, removed.